app/src/main/java/io/heckel/ntfy/up/BroadcastReceiver.kt

```diff
@@ -93,7 +93,12 @@ class BroadcastReceiver : android.content.BroadcastReceiver() {
                 try {
                     // Note, this may fail due to a SQL constraint exception, see https://github.com/binwiederhier/ntfy/issues/185
                     repository.addSubscription(subscription)
-                    distributor.sendEndpoint(appId, connectorToken, endpoint)
+                    /* We don't send the endpoint here anymore, the foreground service will do that after
+                    registering with the push server. This avoids a race condition where the application server
+                    is rejected before ntfy even establishes that this topic exists.
+                    This is fine from an application perspective, because other distributors can't even register
+                    without a connection to the push server.
+                    Unless the app sends registration twice. Then it'll get the endpoint.*/
```

Comment:
^ Pointing this out, hopefully is fine

Reply:
Actually, assuming the Android app is pushed out before the server changes, I'll have to leave the endpoint sending in for now. Otherwise, the new Android app will break for old servers.